### DISCUSSION TITLE
403 error while streaming

### DISCUSSION THREAD

**Discussion by n-szulc** [tweepy/tweepy#1882] [Questions] [answered]
Hello,

I have Elevated Twitter access and would like to use API 1.1 to stream tweets. I have up-to-date keys with read/write/DM permissions and OAuth 1.0a turned on. 

The following code generates ERROR:tweepy.streaming:Stream encountered HTTP error: 403

```
#!/usr/bin/python3

import json
import tweepy
import logging
import datetime
from http.client import IncompleteRead

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger()

keys = {"api": "",
        "api_secret": "",
        "token": "",
        "token_secret": ""
       }

class SimpleListener(tweepy.Stream):

    def on_status(self, status):
        #code to run each time the stream receives a status
        print(status.text)

    def on_error(self, status_code):
        #code to run each time an error is received
        print(status_code)
        if status_code == 420:
            return False
        else:
            return True

def main(keywords):
    start_time = datetime.datetime.now()
    tweepy_listener = SimpleListener(keys["api"], keys["api_secret"],keys["token"],keys["token_secret"])
    tweepy_listener.filter(track=keywords)

if __name__ == "__main__":
    main(['#python'])
```

I also tried with the tweepy 3.10.0 and class SimpleListener(tweepy.StreamListener) (I can provide code if needed), but the issue persisted. My authentication keys are valid, since I am able to post Tweets or send DMs.

Of note, similar issue is has been recently described here: 
https://stackoverflow.com/questions/72102683/how-to-solve-stream-encountered-http-error-403-in-python 

Could you please provide solution to this problem?

Kind regards

**Comment by Harmon758** [comment]
Are you using a Twitter Developer App created on or after 2022-04-29?

**Comment by n-szulc** [comment]
Yes, indeed

**Comment by Harmon758** [comment, marked as answer]
Then this is a duplicate of #1877 / #1878.

> New Apps:
> > will not be able to gain access to v1.1 statuses/sample and v1.1 statuses/filter
> 
> https://twittercommunity.com/t/deprecation-announcement-removing-compliance-messages-from-statuses-filter-and-retiring-statuses-sample-from-the-twitter-api-v1-1/170500
> 
> You'll have to use Twitter API v2.
